test/parallel/test-http-client-check-http-token.js

```diff
@@ -0,0 +1,48 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const http = require('http');
+
+const expectedSuccesses = {
+  undefined: 'GET',
+  null: 'GET',
+  'get': 'GET',
+  'post': 'POST'
+};
+
+const methods = Object.keys(expectedSuccesses);
+let requestCount = 0;
+
+const server = http.createServer((req, res) => {
+  requestCount++;
+  res.end();
+
+  if (methods.length === requestCount) {
+    server.close();
+  }
+}).listen(0, test);
+
+function test() {
+  function fail(input) {
+    assert.throws(() => {
+      http.request({ method: input, path: '/' }, common.fail);
+    }, /Method must be a string/);
```

Comment:
If you want to be really strict, this could be `/^TypeError: Method must be a string$/`.